common/softassert/softassert.go

```diff
@@ -6,7 +8,7 @@ import (
 )
 
 // That performs a soft assertion by logging an error if the given condition is false.
-// It is meant to indicate a condition that are always expected to be true.
+// It is meant to indicate a condition is always expected to be true.
```

Comment:
Already in here but "soft assertion" and "always expected to be true" seem contradictory to me.  What is the use case for a soft assertion?

Reply:
Right; that's a long long story. The `soft` comes from the desire to _not_ fail hard but let outside observers (such as testlogger, CICD and Antithesis) raise an alarm when it sees those. So what we didn't want was to hard crash in production. Hence the "soft".

Reply:
You'll see that `That` maps to Antithesis' `Always` (ie always expected to be true).